Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -547,208 +386,170 @@ function Set-FrontDoorRequiredTLSVersion
+            Write-Host $([Constants]::DoubleDashLine)
         }
+        else
+        {
+            Write-Host "'Force' flag is provided. Minimum required TLS Version will be set on the for all Front Door CDN Profiles Domain(s) without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning)
+            Write-Host $([Constants]::SingleDashLine)
+        }       
 
-        Write-Host $([Constants]::SingleDashLine)
-        Write-Host "`[Step 4 of 4] Set minimum required TLS Version for FrontDoors"
-        Write-Host $([Constants]::SingleDashLine)
+        
         # To hold results from the remediation.
-        $customDomainsRemediated = @()
-        $frontDoorsEPSkipped = @()
+        $frontDoorsDomainsRemediated = @()
+        $frontDoorsDomainsSkipped = @()
         
-        $NonCompliantFrontDoorEndpoints | ForEach-Object {
+        $NonCompliantFrontDoorDomains | ForEach-Object {
             $frontdoorEndpoint = $_
             $resourceGroupName = $_.ResourceGroupName
-            $resourceName = $_.ResourceName
+            $resourceName = $_.ProfileName
+            $DomainName = $_.DomainName
             $DomainName = $_.DomainName
-            $CertificateType =$_.CertificateType
             $PrevMinimumTlsVersion = $_.MinimumTlsVersion
             $MinimumTlsVersion = $_.MinimumTlsVersion
-            $isMinTlsVersionSetOnCustomDomain = $_.isMinTLSVersionSetOnCustomDomain
-            $isMinTLSVersionSetOnCustomDomainPostRemediation = $false
             
-            $frontdoorEndpoint | Add-Member -NotePropertyName isMinTLSVersionSetOnCustomDomainPostRemediation -NotePropertyValue $isMinTLSVersionSetOnCustomDomainPostRemediation
+            
             $frontdoorEndpoint | Add-Member -NotePropertyName PrevMinimumTlsVersion -NotePropertyValue $PrevMinimumTlsVersion
 
-            Write-Host "Setting minimum required TLS Version for custom domains of Front Door :  Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host "Setting minimum required TLS Version for domains of Front Door CDN Profiles :  Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
             Write-Host $([Constants]::SingleDashLine)
-            
-            if (-not [System.Convert]::ToBoolean($isMinTlsVersionSetOnCustomDomain))
+           
+            try
             {
-                try
-                {
-                    Write-Host "Setting minimum required TLS version on custom domain : [$DomainName]" -ForegroundColor $([Constants]::MessageType.Info)
-                    Write-Host $([Constants]::SingleDashLine)
+                Write-Host "Setting minimum required TLS version on domain : [$DomainName]" -ForegroundColor $([Constants]::MessageType.Info)
```

Comment:
Maybe merge this statement with above one and convey the information that this domain which belongs to this FrontDoor profile which is in turn belongs to this Resource group is getting remediated.